### PR TITLE
Update react-native and npms on Android and iOS

### PR DESCRIPTION
**Pull request by ahto**
Manual react-native update done based on -> https://react-native-community.github.io/upgrade-helper/?from=0.61.5&to=0.63.3
Other files were updated based on content in this page except for xcode project file.
For that i revert it back to what it was and followed the guide here https://github.com/react-native-community/upgrade-support/issues/13

When jumping between react-native versions, to clear caches, you can run
`rm -Rf node_modules && npm install && npx jetifier && cd android && ./gradlew clean && cd .. && detox build --configuration android.emu.debug && npx react-native start --reset-cache`

## npm outdated

This is as high as it goes without tsc breaking
```
@types/react-native                     0.63.8        0.63.8  0.63.37  ylitse
```

Highest for version 16.x.x
```
@types/react-test-renderer              16.9.0        16.9.0   17.0.0  ylitse
``` 

Prettier update should be done by the guy who controls linting rules.
```
prettier                                1.18.2        1.18.2    2.2.1  ylitse
```

React navigation update look like big code changes. 
```
react-navigation-stack                   2.5.1         2.5.1   2.10.2  ylitse
```

These npms are held back as per guide https://raw.githubusercontent.com/react-native-community/rn-diff-purge/release/0.63.3/RnDiffApp/package.json
```
@babel/core                              7.8.4         7.8.4   7.12.9  ylitse
@babel/runtime                           7.8.4         7.8.4   7.12.5  ylitse
metro-react-native-babel-preset         0.59.0        0.59.0   0.64.0  ylitse
react                                  16.13.1       16.13.1   17.0.1  ylitse
react-test-renderer                    16.13.1       16.13.1   17.0.1  ylitse
```

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -189,18 +191,23 @@ android {
...
+    //noinspection GradleDynamicVersion
     implementation "com.facebook.react:react-native:+"  // From node_modules
-    implementation 'androidx.appcompat:appcompat:1.1.0-rc01'
-    implementation 'androidx.swiperefreshlayout:swiperefreshlayout:1.1.0-alpha02'
-
 
-    // Firebase dependencies
-    implementation "com.google.android.gms:play-services-base:16.1.0"
-    implementation "com.google.firebase:firebase-core:16.0.9"
-    implementation "com.google.firebase:firebase-messaging:18.0.0"
```
Comment (pheis) [inline android/app/build.gradle:202]: Miks nää on poistunut täältä?

File: android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -15,25 +15,12 @@
...
-      </service>
-
-      <service android:name="io.invertase.firebase.messaging.RNFirebaseMessagingService">
-        <intent-filter>
-          <action android:name="com.google.firebase.MESSAGING_EVENT" />
-        </intent-filter>
-      </service>
-
-      <service android:name="io.invertase.firebase.messaging.RNFirebaseBackgroundMessagingService" />
-      <service android:name="io.invertase.firebase.messaging.RNFirebaseBackgroundMessagingService" />
```
Comment (pheis) [inline android/app/src/main/AndroidManifest.xml:31]: Miks nää on poistettu?

File: android/app/src/main/java/com/ylitse/MainApplication.java [hunk 3]
```diff
@@ -28,7 +25,6 @@ public boolean getUseDeveloperSupport() {
         protected List<ReactPackage> getPackages() {
           @SuppressWarnings("UnnecessaryLocalVariable")
           List<ReactPackage> packages = new PackageList(this).getPackages();
-          packages.add(new RNFirebaseMessagingPackage()); // FCM
```
Comment (pheis) [inline android/app/src/main/java/com/ylitse/MainApplication.java:31]: Miks kaikki Firebase jutut poistettu?

File: android/app/build.gradle (hunk 1 above)
Comment (ahto) [reply android/app/build.gradle:202]: npm kutos versiossa nää on siirtyneet sinne npmmien sisälle.
https://rnfirebase.io/migrating-to-v6#removing-native-dependencies

File: android/app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (ahto) [reply android/app/src/main/AndroidManifest.xml:31]: samaa hommaa yllä olevan kanssa.
npm kutos versiossa nää on siirtyneet sinne npmmien sisälle.
https://rnfirebase.io/migrating-to-v6#removing-native-dependencies

File: android/app/src/main/java/com/ylitse/MainApplication.java (hunk 3 above)
Comment (ahto) [reply android/app/src/main/java/com/ylitse/MainApplication.java:31]: https://rnfirebase.io/migrating-to-v6#removing-the-react-native-firebase-packages

**Review by pheis — APPROVED** [review]

**Pull request merged**